src/tree/SlotTreeItemBase.ts

```diff
@@ -65,6 +70,25 @@ export abstract class SlotTreeItemBase extends AzureParentTreeItem<ISiteTreeRoot
     }
 
     public async refreshImpl(): Promise<void> {
+        let runtime: ProjectRuntime | undefined;
+        try {
+            const appSettings: WebSiteManagementModels.StringDictionary = await this.root.client.listApplicationSettings();
+            runtime = convertStringToRuntime(appSettings.properties && appSettings.properties.FUNCTIONS_EXTENSION_VERSION);
+        } catch {
+            // ignore and use default
+        }
+        // tslint:disable-next-line: strict-boolean-expressions
+        this.runtime = runtime || ProjectRuntime.v2;
+
+        // tslint:disable-next-line: no-any
+        let data: any;
+        try {
+            data = await this.root.client.kudu.functionModel.getHostSettings();
```

Comment:
Why are we special casing/expecting this to fail?

Reply:
It's very similar to the app service plan case you just fixed (where the user didn't have permissions). I don't want a failure for this to block deploy